server/mailet/mailetcontainer-camel/src/main/java/org/apache/james/mailetcontainer/impl/JamesMailSpooler.java

```diff
@@ -220,7 +220,8 @@ public void dispose() {
         while (numActive.get() != 0 && stop > System.currentTimeMillis()) {
             try {
                 Thread.sleep(1000);
-            } catch (Exception ignored) {
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
```

Comment:
not ignored anymore?

Reply:
Yes, having an empty catch block is forbidden so I tried to refine what we really want and having an InterruptedException is very unlikely (see https://stackoverflow.com/questions/26856242/interruptedexception-what-causes-it for how it can happen). I then decided to rethrow it as a runtime.